### ISSUE TITLE
terraform AWS EKS 'destroy' failed module.eks.kubernetes_config_map.aws_auth[0] Error: Unauthorized

### ISSUE THREAD

**Issue by alex-beyond-minds** [terraform-aws-modules/terraform-aws-eks#1661]
## Description

at terraform destroy execution it's failing on `module.eks.kubernetes_config_map.aws_auth[0]`
with error:
`Error: Unauthorized`

## Versions

 - Terraform: 1.0.9

 - Provider(s): kubernetes 2.6.1

 - Module: EKS 17.22.0

## Reproduction

Are you using workspaces?: yes
Have you cleared the local cache (see Notice section above)?: yes
List steps in order that led up to the issue you encountered:
create AWS EKS cluster with custom users, run terraform destroy.

### Code Snippet to Reproduce
module "eks" {
  version = "17.22.0"
  source       = "terraform-aws-modules/eks/aws"
  cluster_name = var.cluster_name
  cluster_version = "1.21"
.
.
.
  map_roles                            = var.map_roles
  map_users                            = var.map_users
}

## Expected behavior
Destroy removes EKS cluster 

## Actual behavior
Destroy is failing on deleting aws_auth configmap

**Comment by alex-beyond-minds** [comment]
Hi @daroga0002 

is anything in provided info causing on issue with my code?

BR

**Comment by daroga0002** [comment]
I reviewed logs suspecting this is because token to EKS expiring. From start to executing aws_auth resource there is more than 20 minutes.


Please try add to kubernetes provider something like:
https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs#exec-plugins

**Comment by alex-beyond-minds** [comment]
I have in my config at this moment:
```
provider "kubernetes" {
  host  = data.aws_eks_cluster.cluster.endpoint
  token = module.eks.eks_token
  #var.eks_token
  cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority.0.data)
  # version = "2.5.1"
  exec {
    api_version = "client.authentication.k8s.io/v1alpha1"
    command     = "aws-iam-authenticator"
    args        = ["token", "-i", data.aws_eks_cluster.cluster.name]
    env = {
      name : "AWS_PROFILE"
      value : var.aws_profile
    }
  }
}

**Comment by alex-beyond-minds** [comment]
This is unlikely related with expiration. In case I remove roles and users parameters, I can install and destroy environment with no issues.

**Comment by antonbabenko** [comment]
This issue has been resolved in version 18.0.0 :tada:
